numba/tests/test_array_manipulation.py

```diff
@@ -1250,6 +1250,69 @@ def test_broadcast_arrays_incompatible_shapes_raise_valueerror(self):
                 self.assertIn("shape mismatch: objects cannot be broadcast to a single shape",
                               str(raises.exception))
 
+    def test_readonly_after_flatten(self):
+        # Reproduces Issue #8370
+        def unfold_flatten(x, y):
+            r, c = x.shape
+            a = np.broadcast_to(x, (y, r, c))
+            b = np.swapaxes(a, 0, 1)
+            cc = b.flatten()
+            d = np.reshape(cc, (-1, c))
+            d[y - 1:, :] = d[: 1 - y]
+            return d
+
+        pyfunc = unfold_flatten
+        cfunc = jit(nopython=True)(pyfunc)
+
+        # If issue #8370 is not fixed: This will fail with "NumbaTypeError:
+        # Cannot modify readonly array of type: readonly array(int32, 2d, C)"
```

Comment:
This isn't the message I get - I get:

```
No implementation of function Function(<built-in function setitem>) found for signature:
 
 >>> setitem(readonly array(int64, 2d, C), UniTuple(slice<a:b> x 2), readonly array(int64, 2d, C))
```

I think it'd be OK to just delete this comment rather than update it.